doc/developer/scripting.rst

```diff
@@ -378,64 +541,75 @@ accomplished with scripting.
-   
-   
-   log.info("Evaluating route " .. prefix.network .. " from peer " .. peer.remote_id.string)
-   
-   function on_match (prefix, attrs)
-           log.info("Match")
-           action = RM_MATCH
-   end
-   
-   function on_nomatch (prefix, attrs)
-           log.info("No match")
-           action = RM_NOMATCH
-   end
-   
-   function on_match_and_change (prefix, attrs)
-           action = RM_MATCH_AND_CHANGE
-           log.info("Match and change")
-           attrs["metric"] = attrs["metric"] + 7
-   end
-   
-   special_routes = {
-           ["172.16.10.4/24"] = on_match,
-           ["172.16.13.1/8"] = on_nomatch,
-           ["192.168.0.24/8"] = on_match_and_change,
-   }
+   -- route_match returns table with following keys:
+   --   integer action, required
+   --     resultant status code. Should be one of RM_*
+   --   table attributes, optional
+   --     updated route attributes
+   --
+
+   function route_match(prefix, attributes, peer,
+           RM_FAILURE, RM_NOMATCH, RM_MATCH, RM_MATCH_AND_CHANGE)
+
+           log.info("Evaluating route " .. prefix.network .. " from peer " .. peer.remote_id.string)
    
+           function on_match (prefix, attributes)
+                   log.info("Match")
+                   return {
+                           attributes = RM_MATCH
+                   }
+           end
    
-   if special_routes[prefix.network] then
-           special_routes[prefix.network](prefix, attributes)
-   elseif peer.stats.update_in % 3 == 0 then
-           on_match(prefix, attributes)
-   elseif peer.stats.update_in % 2 == 0 then
-           on_nomatch(prefix, attributes)
-   else
-           on_match_and_change(prefix, attributes)
-   end
-
+           function on_nomatch (prefix, attributes)
+                   log.info("No match")
+                   return {
+                           action = RM_NOMATCH
+                   }
+           end
+
+           function on_match_and_change (prefix, attributes)
+                   log.info("Match and change")
+                   attributes["metric"] = attributes["metric"] + 7
+                   return {
+                           action = RM_MATCH_AND_CHANGE,
+                           attributes = attributes
+                   }
+           end
+
+           special_routes = {
+                   ["172.16.10.4/24"] = on_match,
+                   ["172.16.13.1/8"] = on_nomatch,
+                   ["192.168.0.24/8"] = on_match_and_change,
+           }
+
+
+           if special_routes[prefix.network] then
+                   return special_routes[prefix.network](prefix, attributes)
+           elseif peer.stats.update_in % 3 == 0 then
+                   return on_match(prefix, attributes)
+           elseif peer.stats.update_in % 2 == 0 then
+                   return on_nomatch(prefix, attributes)
+           else
+                   return on_match_and_change(prefix, attributes)
+           end
+    end
```

Comment:
I think we need some docs in the user guide on how to use it.

Like a "hand holding" kinda of guide of all the steps necessary for a user to setup a script and get it working. I might would also move this example there as well and refer back to it here in the dev guide.

Reply:
Hmm yeah. I'd like to beef up the user docs, not in this PR, but while adding the second hook point? It would become more obvious how this would be phrased for non dev users

Reply:
sure, that will be fine